src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -461,11 +461,11 @@ internal async Task<(IReadOnlyList<T> results, string continuationToken)> Execut
                     if (page.Count > 0)
                     {
                         results.AddRange(page);
-                        if (mustNotExceedMaxItemCount && results.Count > feedOptions.MaxItemCount)
+                        if (mustNotExceedMaxItemCount && page.Count > feedOptions.MaxItemCount)
```

Comment:
>feedOptions.MaxItemCount [](http://example.com/codeflow?start=70&length=24)

This thing get reassign in line 453.

In my test I have 20 results, and cosmos can return me only 9.
So I fetch first 9 and then go to all this fetch more loop.

maxCount is 10 (half of what we ask - 20)

currentDesiredCount is 11, and we set that value to be feedOptions.MaxItemCount.

We do another fetch from cosmos and it returns 9 more results.
So total 18 results.

In old code we would take this 18 results compare them to 
feedOptions.MaxItemCount (which is 11) so that means we have 7 extra results.

So we would flush them off.

In reality, 18 results is still smaller than 20, so we need to return them all.

It looks like bug where we don't consider change in feedOptions.MaxItemCount to happen, so I make change to incorporate that.